src/Core/Services/AuthService.cs

```diff
@@ -198,12 +198,32 @@ private async Task<bool> RequirePasswordChangeAsync(string email, string masterP
             return !await _policyService.EvaluateMasterPassword(strength.Value, masterPassword, _masterPasswordPolicy);
         }
 
-        public async Task<AuthResult> LogInPasswordlessAsync(string email, string accessCode, string authRequestId, byte[] decryptionKey, string userKeyCiphered, string localHashedPasswordCiphered)
+        public async Task<AuthResult> LogInPasswordlessAsync(string email, string accessCode, string authRequestId, byte[] decryptionKey, string masterKey, string masterKeyHash)
         {
-            var decKey = await _cryptoService.RsaDecryptAsync(userKeyCiphered, decryptionKey);
-            var decPasswordHash = await _cryptoService.RsaDecryptAsync(localHashedPasswordCiphered, decryptionKey);
-            return await LogInHelperAsync(email, accessCode, Encoding.UTF8.GetString(decPasswordHash), null, null, null, new MasterKey(decKey), null, null,
+            AuthResult response = null;
+            // On SSO flow user is already AuthN
+            if (await this._stateService.IsAuthenticatedAsync())
```

Comment:
⛏️We're not using `this` unless strictly necessary:
```suggestion
            if (await _stateService.IsAuthenticatedAsync())
```